src/edu/stevens/ece/TextEditor.java

```diff
@@ -26,15 +30,79 @@ public TextEditor() throws Exception {
 		editor = new JTextPane();
 		JScrollPane editorScrollPane = new JScrollPane(editor);
 		editor.setDocument(doc = new DefaultStyledDocument());
+		editor.setFont(defaultFont);
 		add(editorScrollPane, BorderLayout.CENTER);
+		addActions();
 		setVisible(true);
 	}
 	public void addActions() {
-		
+		new IrreversibleAction("SAVE") {
+			@Override public void doIt(ActionEvent e) {
+				save();
+			}
+		};
+		new IrreversibleAction("SAVEAS") {
+			@Override public void doIt(ActionEvent e) {
+				saveas();
+			}
+		};
+		new IrreversibleAction("OPEN") {
```

Comment:
I might have misunderstood the concept, but should opening be an irreversible action?